sys/include/xtimer.h

```diff
@@ -573,11 +555,43 @@ int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t us);
 #define XTIMER_MASK (0)
 #endif
 
+#define XTIMER_HZ_BASE (1000000ul)
+
 #ifndef XTIMER_HZ
 /**
  * @brief  Frequency of the underlying hardware timer
  */
-#define XTIMER_HZ 1000000ul
+#define XTIMER_HZ XTIMER_HZ_BASE
+#endif
+
+#ifndef XTIMER_SHIFT
+#if (XTIMER_HZ >> 1 == XTIMER_HZ_BASE) || (XTIMER_HZ << 1 == XTIMER_HZ_BASE)
+#define XTIMER_SHIFT (1)
+#elif (XTIMER_HZ >> 2 == XTIMER_HZ_BASE) || (XTIMER_HZ << 2 == XTIMER_HZ_BASE)
+#define XTIMER_SHIFT (2)
+#elif (XTIMER_HZ >> 3 == XTIMER_HZ_BASE) || (XTIMER_HZ << 3 == XTIMER_HZ_BASE)
+#define XTIMER_SHIFT (3)
+#elif (XTIMER_HZ >> 4 == XTIMER_HZ_BASE) || (XTIMER_HZ << 4 == XTIMER_HZ_BASE)
+#define XTIMER_SHIFT (4)
+#else
+/**
+ * @brief   xtimer prescaler value
+ *
+ * If the underlying hardware timer is running at a power of two multiple of
+ * 15625, XTIMER_SHIFT can be used to adjust the difference.
+ *
+ * For a 1 MHz hardware timer, set XTIMER_SHIFT to 0.
+ *
+ * For a 4 MHz hardware timer, set XTIMER_SHIFT to 2.
+ * For a 16 MHz hardware timer, set XTIMER_SHIFT to 4.
+ * For a 250 kHz hardware timer, set XTIMER_SHIFT to 2.
+ *
+ * The direction of the shift is handled by the macros in tick_conversion.h
+ */
+#define XTIMER_SHIFT (0)
```

Comment:
good point, will adapt. Btw. 500kHz would be `XTIMER_SHIFT (1)` which is covered above, but not mentioned in the docu 😕